### PR TITLE
Add horizontal pod autoscaler and pod disruption budgets

### PR DESCRIPTION
**Pull request by andytson-inviqa**
autoscaler will be added based on .Values.services.*.autoscaling config,
and pod disruption budgets default to minAvailable 1 only if minimum replicas is greater than 1

### CODE REVIEW

**Review by kierenevans — DISMISSED** [review]

File: src/_base/helm/app/templates/pod-disruption-budgets.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{- range $serviceName, $service := omit .Values.services "php-base" -}}
+{{- with $service }}
+{{- if and (not (hasPrefix "." $serviceName)) .enabled (hasKey . "disruptionBudget" -}}
```
Comment (kierenevans) [inline src/_base/helm/app/templates/pod-disruption-budgets.yaml]:
```suggestion
{{- if and (not (hasPrefix "." $serviceName)) .enabled (hasKey . "disruptionBudget") -}}
```

**Review by kierenevans — DISMISSED** [review]

File: src/_base/helm/app/templates/pod-disruption-budgets.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{{- range $serviceName, $service := omit .Values.services "php-base" -}}
+{{- with $service }}
+{{- if and (not (hasPrefix "." $serviceName)) .enabled (hasKey . "disruptionBudget") -}}
```
Comment (andytson-inviqa) [inline src/_base/helm/app/templates/pod-disruption-budgets.yaml]:
```suggestion
{{- if and (not (hasPrefix "." $serviceName)) .enabled (hasKey . "disruptionBudget") }}
```

**Review by kierenevans — DISMISSED** [review]

**Comment by andytson-inviqa** [comment]
I'm going to change it to have a sane minAvailable: 1 default and only used if replicas > 1 or hpa and minReplicas > 1

File: src/_base/helm/app/templates/horizontal-pod-autoscaler.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    argocd.argoproj.io/sync-wave: "15"
+spec:
+  {{- with (omit $autoscaling "enabled" "metrics") }}
+  {{- . | toYaml | nindent 2 }}
+  {{- end }}
+  metrics:
+    {{- with $autoscaling.metrics }}
+    {{- . | toYaml | nindent 4 }}
+    {{- end }}
+    {{- with $autoscaling.targetMemoryUtilizationPercentage }}
```
Comment (andytson-inviqa) [inline src/_base/helm/app/templates/horizontal-pod-autoscaler.yaml:24]:
```suggestion
    {{- /* the following is not recommended for multi-container pods due to averaging of metrics, use .metrics instead https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics */ }}
    {{- with $autoscaling.targetMemoryUtilizationPercentage }}
```

File: src/_base/helm/app/templates/horizontal-pod-autoscaler.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    argocd.argoproj.io/sync-wave: "15"
+spec:
+  {{- with (omit $autoscaling "enabled" "metrics") }}
+  {{- . | toYaml | nindent 2 }}
+  {{- end }}
+  metrics:
+    {{- with $autoscaling.metrics }}
+    {{- . | toYaml | nindent 4 }}
+    {{- end }}
+    {{- /* the following is not recommended for multi-container pods due to averaging of metrics, use .metrics instead https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics */ }}
```
Comment (andytson-inviqa) [inline src/_base/helm/app/templates/horizontal-pod-autoscaler.yaml]:
```suggestion
```

**Review by marcomc — APPROVED** [review]
all looking good

**Pull request merged**